File: app/templates/server_node/koa/server_node_koa_babel/api/todo/controller/todo-controller.js
"use strict";

import TodoDAO from "../dao/todo-dao";

export default class TodoController {
  async getAll() {
      try {
        let _todos = await TodoDAO.getAll();
        this.status = 200;
        this.body = _todos;
      } catch(e) {
        this.status = 400;
      }
  }

  async getById() {
      try {
        let _todo = await TodoDAO.getById(this.param.id);
        this.status = 200;
        this.body = _todo;
      } catch(e) {
        this.status = 400;
      }
  }

  async createTodo() {
      let _todo = this.request.body;

      try {
        let _newTodo = await TodoDAO.createTodo(_todo);
        this.body = _newTodo;
        this.status = 201;
      } catch(e) {
        this.status = 400;
      }
  }

  async deleteTodo() {
    let _id = this.params.id;

    try {
      await TodoDAO.deleteTodo(_id);
      this.status = 200;
    } catch(e) {
      this.status = 400;
    }
  }
}

File: app/templates/server/go/gin/api/todo/controller/todocontroller.go
package todocontroller

import (
	"<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/todo/dao"
	todo "<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/todo/model"
	"github.com/gin-gonic/gin"
	"net/http"
)

func GetAll(c *gin.Context) {
	ts, _ := tododao.All()

	c.JSON(http.StatusOK, ts)
}

func GetById(c *gin.Context) {
	id := c.Param("id")

	nt, _ := tododao.GetById(id)

	c.JSON(http.StatusOK, nt)
}

func NewTodo(c *gin.Context) {
	t := new(todo.Todo)

	c.Bind(t)

	nt, _ := tododao.NewTodo(*t)

	c.JSON(http.StatusOK, nt)
}

func RemoveTodo(c *gin.Context) {
	id := c.Param("id")

	tododao.DeleteTodo(id)

	c.String(http.StatusOK, "")
}
